client/state/sites/domains/schema.js

```diff
@@ -0,0 +1,37 @@
+export const itemsSchema = {
+	type: 'array',
```

Comment:
Looking at your data structure, the array should be one level down.

This is currently testing for:
```
[ { '123456': { 'domain': 'foo.com' } } ]
```
vs
```
{ '123456': [ { 'domain': 'foo.com' }, { 'domain': 'bar.com' } ] }
```